python/ray/serve/_private/controller.py

```diff
@@ -929,10 +929,15 @@ def get_serve_instance_details(self) -> Dict:
         # Eventually we want to remove route_prefix from DeploymentSchema.
         http_options = HTTPOptionsSchema.parse_obj(http_config.dict(exclude_unset=True))
         grpc_options = gRPCOptionsSchema.parse_obj(grpc_config.dict(exclude_unset=True))
+        proxy_location = (
+            None
+            if http_config.location is None
+            else ProxyLocation._from_deployment_mode(http_config.location)
```

Comment:
Non-blocker: I feel we can just have an early return in `_from_deployment_mode` to return `None` when `None` is passed so we don't need this if else logics here :)

Reply:
Sounds good, I added an early return.